### PR TITLE
Clarify connector support for SQL MERGE

### PR DESCRIPTION
**Pull request by jhlodin**
## Description

`MERGE` support may be misleading, since the source table can come from any catalog. Update documentation to be more explicit about SQL `MERGE` support.

> Is this change a fix, improvement, new feature, refactoring, or other?

Improvement

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

Documentation

> How would you describe this change to a non-technical end user or system administrator?

Be more clear about what it means when a connector supports or does not support `MERGE`

## Documentation

( ) No documentation is needed.
(x) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(x) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Review by djsstarburst — APPROVED** [review]
👍

File: docs/src/main/sphinx/sql/merge.rst [hunk 1]
```diff
@@ -93,5 +93,7 @@ table row::
 Limitations
 -----------
 
-Some connectors have limited or no support for ``MERGE``.
-See connector documentation for more details.
+While any connector can provide a ``MERGE`` source table, if a connector
+does not support ``MERGE`` statements it cannot be the target of a merge
+operation. See the :doc:`connector documentation </connector>` for more
+information.
```
Comment (hashhar) [inline docs/src/main/sphinx/sql/merge.rst:99]: Is this better?

```rst
While any connector can be used as a source table for a ``MERGE`` statement not all connectors can
be the target of a merge operation. Only connectors which support the ``MERGE`` statement can be
the target of a merge operation. See the :doc:`connector documentation </connector>` for more
information.
```

Comment (hashhar) [reply docs/src/main/sphinx/sql/merge.rst:99]: the 1st and 2nd sentences can be combined or reworded to something like:

```rst
Any connector can be used as a source table for a ``MERGE`` statement.
Only connectors which support the ``MERGE`` statement can be the target of a merge operation.
See the :doc:`connector documentation </connector>` for more
information.
```

Comment (mosabua) [reply docs/src/main/sphinx/sql/merge.rst:99]: I like the second suggestion.. starting with "While" seems semi-ideal

**Review by mosabua — CHANGES_REQUESTED** [review]

**Review by hashhar — APPROVED** [review]

**Review by mosabua — APPROVED** [review]

**Pull request merged**